server/semantics-service/src/main/java/edu/illinois/ncsa/incore/service/semantics/controllers/TypeController.java

```diff
@@ -263,42 +271,55 @@ public Response searchType(
     @Operation(summary = "Publish new type.")
     public Response publishType(
         @Parameter(name = "Type uri (name).") Document type) {
-        Space space = spaceRepository.getSpaceByName(this.username);
-
-        String id = this.typeDAO.postType(type);
-
-        // add id to matching space
-        space.addMember(id);
-        spaceRepository.addSpace(space);
-
-        return Response.ok(id).status(200)
-            .build();
+        try {
+            if (authorizer.isUserAdmin(this.groups)) {
+                Space space = spaceRepository.getSpaceByName(this.username);
+                String id = this.typeDAO.postType(type);
+                // add id to matching space
+                space.addMember(id);
+                spaceRepository.addSpace(space);
 
+                return Response.ok(id).status(200)
```

Comment:
it might be nicer to return the whole post body as response instead of just id. 
but you might have to change the return of `this.typeDAO.postType(type)`

Reply:
Thanks for pointing this out! I improved the responses a bit.

Reply:
I made this change: 
After POST successfully, it will return the new type instead of an id